backend/src/main/java/com/google/sps/servlets/AddParticipantServlet.java

```diff
@@ -99,25 +87,48 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       return;
     }
 
-    // id is irrelevant, only relevant when getting from datastore
+    // Create new Participant from input parameters
     Participant newParticipant =
         new Participant(
-            /* id= */ -1L, username, startTimeAvailable, endTimeAvailable, duration, timestamp);
+            username, startTimeAvailable, endTimeAvailable, duration, /* matchId=*/ 0, timestamp);
 
+    // Get DatastoreService and instiate Match and Participant Datastores
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    MatchDatastore matchDatastore = new MatchDatastore(datastore);
+    ParticipantDatastore participantDatastore = new ParticipantDatastore(datastore);
+
+    // Check if new participant already in datastore (unmatched, in queue)
+    if (participantDatastore.getParticipantFromUsername() == null) {
```

Comment:
`participantDatastore.getParticipantFromUsername(username) == null`

Reply:
Oh, just realized that this method call without the param is causing the build to fail

Reply:
Yup fixed it, just forgot to mvn test before